iOS/TwitterImagePipeline/source/TwitterImagePipeline/ApiDefinition.cs

```diff
@@ -0,0 +1,2147 @@
+		[Field("TIPProblemImageDownloadedHasGPSInfo", "__Internal")]
+		NSString ImageDownloadedHasGpsInfo { get; }
+	}
+
+	[Static]
+	partial interface TIPProblemInfoKeys
+	{
+		// extern NSString *const _Nonnull TIPProblemInfoKeyImageIdentifier;
+		[Field("TIPProblemInfoKeyImageIdentifier", "__Internal")]
+		NSString ImageIdentifier { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeySafeImageIdentifier;
+		[Field("TIPProblemInfoKeySafeImageIdentifier", "__Internal")]
+		NSString SafeImageIdentifier { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyImageURL;
+		[Field("TIPProblemInfoKeyImageURL", "__Internal")]
+		NSString ImageUrl { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyTargetDimensions;
+		[Field("TIPProblemInfoKeyTargetDimensions", "__Internal")]
+		NSString TargetDimensions { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyTargetContentMode;
+		[Field("TIPProblemInfoKeyTargetContentMode", "__Internal")]
+		NSString TargetContentMode { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyScaledDimensions;
+		[Field("TIPProblemInfoKeyScaledDimensions", "__Internal")]
+		NSString ScaledDimensions { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyImageDimensions;
+		[Field("TIPProblemInfoKeyImageDimensions", "__Internal")]
+		NSString ImageDimensions { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyImageIsAnimated;
+		[Field("TIPProblemInfoKeyImageIsAnimated", "__Internal")]
+		NSString ImageIsAnimated { get; }
+
+		// extern NSString *const _Nonnull TIPProblemInfoKeyFetchRequest;
+		[Field("TIPProblemInfoKeyFetchRequest", "__Internal")]
+		NSString FetchRequest { get; }
+	}
+
+	[Static]
+	partial interface TIPGlobalConfigurationDefaults
+	{
+		// extern const SInt64 TIPMaxBytesForAllRenderedCachesDefault;
+		[Field("TIPMaxBytesForAllRenderedCachesDefault", "__Internal")]
+		long MaxBytesForAllRenderedCaches { get; }
+
+		// extern const SInt64 TIPMaxBytesForAllMemoryCachesDefault;
+		[Field("TIPMaxBytesForAllMemoryCachesDefault", "__Internal")]
+		long MaxBytesForAllMemoryCaches { get; }
+
+		// extern const SInt64 TIPMaxBytesForAllDiskCachesDefault;
+		[Field("TIPMaxBytesForAllDiskCachesDefault", "__Internal")]
+		long MaxBytesForAllDiskCaches { get; }
+
+		// extern const NSInteger TIPMaxConcurrentImagePipelineDownloadCountDefault;
+		[Field("TIPMaxConcurrentImagePipelineDownloadCountDefault", "__Internal")]
+		nint MaxConcurrentImagePipelineDownloadCount { get; }
+
+		// extern const NSTimeInterval TIPMaxEstimatedTimeRemainingForDetachedHTTPDownloadsDefault;
+		[Field("TIPMaxEstimatedTimeRemainingForDetachedHTTPDownloadsDefault", "__Internal")]
+		double MaxEstimatedTimeRemainingForDetachedHttpDownloads { get; }
+
+		// extern const NSUInteger TIPMaxRatioSizeOfCacheEntryDefault;
+		[Field("TIPMaxRatioSizeOfCacheEntryDefault", "__Internal")]
+		nuint MaxRatioSizeOfCacheEntry { get; }
+
+		// TODO: this is really a short
```

Comment:
Currently, this is not yet supported by the binding engine: https://bugzilla.xamarin.com/show_bug.cgi?id=58845

But, I did it manually.